langchain/src/vectorstores/azuresearch.ts

```diff
@@ -0,0 +1,562 @@
+import * as uuid from "uuid";
+import { SearchClient, SearchIndexClient, AzureKeyCredential, IndexingResult, SearchIndex } from "@azure/search-documents";
+import { VectorStore } from "./base.js";
+import { Embeddings } from "../embeddings/base.js";
+import { Document } from "../document.js";
+
+/**
+ * Azure Search - Type of query.
+ */
+export type AzureSearchType = "semantic_hybrid" | "similarity_hybrid" | "similarity";
+
+/**
+ * Azure search - Store DB Config.
+ */
+export type AzureSearchStoreParams = {
+  client: {
+    indexName: string;
+    endpoint: string;
+    credential: string;
+  } | SearchClient<AzureSearchDocument>;
+  search: {
+    type: AzureSearchType;
+    language?: string;
+    semantic?: string;
+  };
+}
+
+/**
+ * Define metadata schema.
+ *
+ * If yout want to add custom data, use the attributes property.
+ */
+export type AzureSearchDocumentMetadata = {
+  source: string;
+  attributes?: Array<{ key: string; value: string; }>;
```

Comment:
I don't love this restriction as many advanced retrievers will rely on being able to set metadata as they please - do we think we could just do the mapping from `AzureSearchDocumentMetadata.attributes` to arbitrary metadata?

Reply:
Or what is the purpose of `source` here?

Reply:
Thinking about how the developer will use that feature, I used the attributes field to prevent developer need to specify the Schema of your Document (like python langchain implementation does) because Cognitive Search has no Dynamic field.

But the pattern used on all others vector stores is not have that. Your suggestion about create a map é good, I will try to implement that.

Reply:
> Or what is the purpose of `source` here?

In my case the field 'source' is required because is the description about where the document is from. I'll remove this constraint.